backends/cuda-gen/ceed-cuda-gen-operator.c

```diff
@@ -125,7 +125,7 @@ static int CeedOperatorApply_Cuda_gen(CeedOperator op, CeedVector invec,
     ierr = CeedRunKernelDimSharedCuda(ceed, data->op, grid, Q1d, Q1d, elemsPerBlock,
                                       sharedMem, opargs);
   } else if (dim==3) {
-    const CeedInt elemsPerBlock = Q1d<8? 4 : 1;
+    const CeedInt elemsPerBlock = Q1d<6? 4 : (Q1d<8? 2 : 1);
```

Comment:
I'm not sure. This heuristic was found experimentally on BP3, it could certainly be improved for complex operators later. The idea is mostly to have a number of active threads per block as close as possible to a warp size (multiple of 32), not so much about the amount of resources used per thread. The number of components mostly impact the number of registers used by each threads, so I'm not sure it's relevant here. It would be relevant if we could choose between different implementations (that use more or less registers, shared mem, etc), but I only have one implementation in cuda-gen for now. I thought about unifying all cuda backends so that I can use the code in cuda-gen with any of the previous implementation (it's mostly using the functions from cuda-shared reformated for cuda-gen). This would require time with very uncertain interest, so it's not very high in my priority list for the moment.